### DISCUSSION TITLE
CI blocks after a commit is pushed by github workflow

### DISCUSSION THREAD

**Discussion by pysel** [community/community#50468] [Actions] [answered]
### Select Topic Area

Question

### Body

I have a workflow that is updating the dependencies of a go project, when a push to a local branch happens, and pushes these changes to the same branch. However, if it happens that these automatic pushes are made to the branch with an opened PR, the CI blocks in the PR with `Expected — Waiting for status to be reported`. It can be solved by manual closing/reopening a PR, but that is a not desirable overhead for developers. 

I was wondering if anyone knows the reason for that. [This](https://github.com/osmosis-labs/osmosis/blob/main/.github/workflows/go-mod-auto-bump.yml) is the workflow that is pushing some changes with updated dependencies to the local branch, and this:

<img width="947" alt="Screen Shot 2023-03-20 at 10 47 06" src="https://user-images.githubusercontent.com/46343690/226242003-ac663321-6f7a-492b-90be-8b9bf248f309.png">

is how CI looks after an [auto-commit](https://github.com/osmosis-labs/osmosis/pull/4667/commits/5bcf1d463e330bb373d8b6f4783de0cfb49e129b) is made

**Comment by airtower-luna** [comment, marked as answer]
The problem is that the push is done using the `GITHUB_TOKEN`, which [won't create new workflow runs](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow), so the required checks never happen for the automatically created commit. If you want the push to create new workflow runs, you need to use [a PAT instead](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) to authorized the push (generally the best way to do it is to [pass it to `actions/checkout` using its `token` parameter](https://github.com/actions/checkout#usage)).

**Reply by pysel** [reply]
thank you @airtower-luna! I have just another question, just to learn: what permissions does `PAT` grant that `GITHUB_TOKEN` does not, that it allows CI not to block?

**Reply by pysel** [reply]
for example, in the link you shared above, there is a [github token permissions topic](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token), which covers several scopes and corresponding accesses. Another way to put my question, I guess, is: which scope is responsible for CI blocking/not blocking after a workflow push?

**Reply by airtower-luna** [reply]
It's not a scope, it's a special case for the `GITHUB_TOKEN` by design. To quote the documentation I linked above:

> When you use the repository's `GITHUB_TOKEN` to perform tasks, events triggered by the `GITHUB_TOKEN`, with the exception of `workflow_dispatch` and `repository_dispatch`, will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs. For example, if a workflow run pushes code using the repository's `GITHUB_TOKEN`, a new workflow will not run even when the repository contains a workflow configured to run when `push` events occur.

**Reply by pysel** [reply]
thanks for the response @airtower-luna! The reason I asked this question is more about understanding what specific permission (lets call it like that) a token has to have in order not to block CI. I am working on this workflow for a repository that has a few secrets, however, when using them in the workflow, the CI still blocks.

So, I will try to rephrase my question: what specific permissions does the `PAT` grants that the CI is not blocking when you use it? Does it basically mean that using `PAT` is pretty much equal to authenticating as a user, on behalf of who the commit is pushed, which is the requirement for the CI not to block? If so, is there a way to commit on behalf of a github action bot without blocking a CI? Let me know if that makes sense, please, thank you

**Reply by airtower-luna** [reply]
> Does it basically mean that using PAT is pretty much equal to authenticating as a user, on behalf of who the commit is pushed, which is the requirement for the CI not to block?

Pretty much, yes. Except there's nothing blocking as such, the `GITHUB_TOKEN` just can't trigger workflow runs for the newly pushed commit. Which then implicitly blocks the merge if you have a branch protection rule that requires those runs.

> If so, is there a way to commit on behalf of a github action bot without blocking a CI?

If you mean pushing without a PAT, no. If you mean having the actions bot show as the commit author, that should work if you set [author name and email as shown in the `actions/checkout` docs](https://github.com/actions/checkout#push-a-commit-using-the-built-in-token).

**Reply by pysel** [reply]
thanks!
